src/CMakeLists.txt

```diff
@@ -234,6 +234,8 @@ else()
 	endif()
 endif()
 
+target_link_libraries(shared PUBLIC CLI11::CLI11)
```

Comment:
Please add the dependency to the client and the server explicitly - the `shared` target really should only have dependencies for the shared code and not dependencies that both components happen to create in their own code respectively.